### PR TITLE
NVSHAS-9243 support deep UI integration, fetch workload scan result b…

### PR DESCRIPTION
**Pull request by jeffhuang4704**
…y container id list

### CODE REVIEW

File: controller/rest/workload.go [hunk 1]
```diff
@@ -262,7 +262,21 @@ func handlerWorkloadListBase(apiVer string, w http.ResponseWriter, r *http.Reque
...
+	// get POST body
+	idlist := utils.NewSet()
+	if r.Method == http.MethodPost {
+		var idListData api.RESTAssetIDList
+		body, _ := io.ReadAll(r.Body)
+		err := json.Unmarshal(body, &idListData)
+		if err != nil {
+			// TODO: error handling
+			log.WithFields(log.Fields{"error": err}).Error("Request error")
+			restRespError(w, http.StatusBadRequest, api.RESTErrInvalidRequest)
```
Comment (williamlin-suse) [inline controller/rest/workload.go:274]: should it return after restRespError(w, http.StatusBadRequest, api.RESTErrInvalidRequest) is called?

File: controller/rest/workload.go [hunk 2]
```diff
@@ -262,7 +262,21 @@ func handlerWorkloadListBase(apiVer string, w http.ResponseWriter, r *http.Reque
 		return
 	}
 
-	cached := cacher.GetAllWorkloads(view, acc)
+	// get POST body
+	idlist := utils.NewSet()
+	if r.Method == http.MethodPost {
```
Comment (williamlin-suse) [inline controller/rest/workload.go:267]: Could query.brief be true when POST("/v2/workload") is requested?
Then the payload is ignored & returns in line 232?

File: controller/rest/workload.go
Comment (williamlin-suse) [inline controller/rest/workload.go:1]: POST("/v2/workload") is actually for user with read permission.
Should line 238 in be handlerWorkloadListBase()
acc, login := getAccessControl(w, r, access.AccessOPRead) 
?

File: controller/rest/workload.go (hunk 1 above)
Comment (jeffhuang4704) [reply controller/rest/workload.go:274]: yes, you're right.
fixed.

File: controller/rest/workload.go
Comment (jeffhuang4704) [reply controller/rest/workload.go:1]: fixed.

File: controller/rest/workload.go (hunk 2 above)
Comment (jeffhuang4704) [reply controller/rest/workload.go:267]: discussed with the use case with UI team. The query.brief will not be used in this case.

**Review by williamlin-suse — APPROVED** [review]

**Pull request merged**